src/vmm/src/builder.rs

```diff
@@ -441,6 +442,65 @@ pub fn build_microvm_for_boot(
     Ok(vmm)
 }
 
+/// Error type for [`build_microvm_from_snapshot`].
+#[derive(Debug, thiserror::Error)]
+pub enum BuildMicrovmFromSnapshotError {
```

Comment:
I like having error types be closely bound to the function/s they correspond to, I find it increases readability.

I think the issue here is rather that `builder.rs` is just too big to begin with, potentially some functions could be moved into sub-modules (although I think this may be outside the scope of this PR).